src/asm/x86/predict.rs

```diff
@@ -131,13 +134,27 @@ pub fn dispatch_predict_intra<T: Pixel>(
         | PredictionMode::D157_PRED
         | PredictionMode::D203_PRED
         | PredictionMode::D67_PRED => {
-          (if angle <= 90 {
-            rav1e_ipred_z1_avx2
-          } else if angle < 180 {
-            rav1e_ipred_z2_avx2
+          let enable_ief = ief_params.is_some() as libc::c_int;
+
+          if enable_ief > 0 {
+            // FIXME: segfaults with smooth filter, desyncs without
+            call_native(dst);
           } else {
-            rav1e_ipred_z3_avx2
-          })(dst_ptr, stride, edge_ptr, w, h, angle);
+            let ief_smooth_filter = if let Some(params) = ief_params {
+              params.use_smooth_filter()
+            } else {
+              false
+            } as libc::c_int;
+            let angle = angle | (enable_ief << 10) | (ief_smooth_filter << 9);
```

Comment:
```suggestion
            let angle_with_flg = angle | (enable_ief << 10) | (ief_smooth_filter << 9);
```
This and update in L156 fixes segfaults with smooth filter but still with desync.

A comment explaining enable_ief and ief_smooth_filter is encoded in bit of angle will help.

Reply:
Done.